core/src/main/scala/org/apache/spark/util/JsonProtocol.scala

```diff
@@ -382,7 +382,8 @@ private[spark] object JsonProtocol {
 
   def executorInfoToJson(executorInfo: ExecutorInfo): JValue = {
     ("Host" -> executorInfo.executorHost) ~
-    ("Total Cores" -> executorInfo.totalCores)
+    ("Total Cores" -> executorInfo.totalCores) ~
+    ("Log Urls" -> mapToJson(executorInfo.logUrlMap))
```

Comment:
should this use `Utils.jsonOption` per the comment at the top of `JsonProtocol` when adding new fields?  Also I guess there should be a backward compatibility test case added to `JsonProtocolSuite`